tests/phpunit/integration/API/Packs/ApiLabkiPacksApplyTest.php

```diff
@@ -0,0 +1,557 @@
+
+use ApiTestCase;
+use LabkiPackManager\Services\LabkiRepoRegistry;
+use LabkiPackManager\Services\LabkiRefRegistry;
+use LabkiPackManager\Services\LabkiPackRegistry;
+use LabkiPackManager\Services\LabkiPageRegistry;
+use LabkiPackManager\Services\LabkiOperationRegistry;
+use LabkiPackManager\Domain\OperationId;
+use MediaWiki\MediaWikiServices;
+
+/**
+ * Integration tests for ApiLabkiPacksApply.
+ *
+ * These tests verify that the unified API endpoint:
+ * - Creates operation records in LabkiOperationRegistry
+ * - Queues LabkiPackApplyJob with correct parameters
+ * - Returns appropriate responses for various scenarios
+ * - Validates input parameters correctly (repo, ref, operations)
+ * - Handles mixed operations (install + update + remove)
+ * - Validates dependencies across operations
+ *
+ * @covers \LabkiPackManager\API\Packs\ApiLabkiPacksApply
+ * @covers \LabkiPackManager\API\Packs\PackApiBase
+ * @group LabkiPackManager
+ * @group Database
+ * @group API
+ */
+class ApiLabkiPacksApplyTest extends ApiTestCase {
+
+	private LabkiRepoRegistry $repoRegistry;
+	private LabkiRefRegistry $refRegistry;
+	private LabkiPackRegistry $packRegistry;
+	private LabkiPageRegistry $pageRegistry;
+	private LabkiOperationRegistry $operationRegistry;
+	private string $testWorktreePath;
+
+	/** @var string[] Tables used by this test */
+	protected $tablesUsed = [
+		'labki_content_repo',
+		'labki_content_ref',
+		'labki_pack',
+		'labki_page',
+		'labki_pack_dependency',
+		'labki_operations',
+		'job',
+	];
+
+	protected function setUp(): void {
+		parent::setUp();
+		
+		$this->repoRegistry = new LabkiRepoRegistry();
+		$this->refRegistry = new LabkiRefRegistry();
+		$this->packRegistry = new LabkiPackRegistry();
+		$this->pageRegistry = new LabkiPageRegistry();
+		$this->operationRegistry = new LabkiOperationRegistry();
+		
+		// Grant permissions for testing
+		$this->setGroupPermissions( 'user', 'labkipackmanager-manage', true );
+
+		// Create temporary directory for test worktree
+		$this->testWorktreePath = sys_get_temp_dir() . '/labki_api_apply_test_' . uniqid();
+		mkdir( $this->testWorktreePath, 0777, true );
+	}
+
+	protected function tearDown(): void {
+		// Clean up test worktree
+		if ( is_dir( $this->testWorktreePath ) ) {
+			$this->recursiveDelete( $this->testWorktreePath );
+		}
+		
+		parent::tearDown();
+	}
+
+	/**
+	 * Recursively delete a directory.
+	 */
+	private function recursiveDelete( string $dir ): void {
+		if ( !is_dir( $dir ) ) {
+			return;
+		}
+
+		$files = array_diff( scandir( $dir ), [ '.', '..' ] );
+		foreach ( $files as $file ) {
+			$path = $dir . '/' . $file;
+			is_dir( $path ) ? $this->recursiveDelete( $path ) : unlink( $path );
+		}
+		rmdir( $dir );
+	}
+
+	/**
+	 * Create a test manifest file in the worktree.
+	 */
+	private function createTestManifest( array $packs = [], array $pages = [] ): void {
+		$yaml = "schema_version: 1\n";
+		
+		// Add packs
```

Comment:
probably construct this as an object and then dump to yaml rather than manually constructing as a string